main/src/ca/uwaterloo/flix/language/phase/Inliner.scala

```diff
@@ -129,6 +129,7 @@ object Inliner {
     case Expression.JumpTo(_, _, _) => exp0
 
     case Expression.Let(sym, exp1, exp2, occur, tpe, purity, loc) =>
+      // Wants to inline `exp1` if `occur` is Once and `purity` is Pure
```

Comment:
I think you should write a few block comments that describe each case, e.g.:

///
/// Case 1: If an expression occurs once and its pure then it is safe to inline without increasing neither the code size nor the execution time.
///

and so forth.